### DISCUSSION TITLE
Is there a way to force vsomeip to never use unix domain sockets for communication, and instead use UDP by default?

### DISCUSSION THREAD

**Discussion by lpcvoid** [COVESA/vsomeip#476] [Q&A]
Hi, title says it all. Is there a way to get vsomeip to not use Unix Domain Sockets for a given service? The current behavior is that vsomeip will transport method calls via Unix Domain Sockets (unless configured as reliable -> TCP) if it determines that both client and server applications are on the same host.

Ideally, there would be a way to configure this behavior via the franca fdepl files for a given service.

The background is that we wish to be able to debug payloads of services with tcpdump in all cases. This isn't possible if applications communicate via Unix Domain Sockets. We are aware that this may cause more overhead within the system in total.

Any way to easily do this?

**Comment by oncaphillis** [comment]
If you can afford to use a special debug version and don't need it permanently this might be the lines which could help you.
https://github.com/COVESA/vsomeip/blob/6e699b57554165dfd10375e034094b20baa3f195/implementation/endpoints/src/server_endpoint_impl.cpp#L172
https://github.com/COVESA/vsomeip/blob/6e699b57554165dfd10375e034094b20baa3f195/implementation/endpoints/src/client_endpoint_impl.cpp#L187
This should log any datagram regardless of UDP or UNIX SOCK
But as you pointed out, it's costly

**Reply by lpcvoid** [reply]
Hi Sebastian, thanks for the answer ;) The problem is that in our particular case, we cannot really use the special vsomeip version since we sometimes need to debug field return units. In that case, having communication run over UDP (which could be mirrored to the outside via diagnostic routine) is just more ergonomic than swapping vsomeip binary on sealed samples.

**Reply by oncaphillis** [reply]
Well both solutions (UDP and tracing) are not really appropriate for release code anyway. The only way I can think of would be to enable the #defed out code via an env variable. That is as good as having a config flag. But in all cases it will be none release code anyway.
And I didn't find any config flag that would enable mandatory UDP com.

**Reply by duartenfonseca** [reply]
hi @lpcvoid , the only possible solution is to always use tcp as internal communication, which is set via configuration

**Reply by maikel** [reply]
Hi @duartenfonseca. Can you point us to the configuration?

**Reply by duartenfonseca** [reply]
try this: [local_tcp.zip](https://github.com/user-attachments/files/19589017/local_tcp.zip)
in the same machine, run the routingmanagerd using the service config, and then one app as subscribe-sample with the client config and one app as notify-sample using the service config

keep the main unicast address as the loopback address, and change only the
```
 "routing": {
        "host": {
            "name": "routingmanagerd",
            "unicast": "172.17.0.4",
            "port": "31000"
        },
        "guests": {
            "unicast": "172.17.0.4",
            "ports": [
            {
                "first": 1026,
                "last": 65535
            }
            ]
        }
```
part to the desired address
